src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -23,23 +25,27 @@ public class SecretManager : IDisposable, ISecretManager
         private readonly ISecretsRepository _repository;
         private HostSecretsInfo _hostSecrets;
         private SemaphoreSlim _hostSecretsLock = new SemaphoreSlim(1, 1);
+        private IMetricsLogger _metricsLogger;
+        private string _repositoryClassName;
 
         // for testing
         public SecretManager()
         {
         }
 
-        public SecretManager(ISecretsRepository repository, ILogger logger, bool createHostSecretsIfMissing = false)
-            : this(repository, new DefaultKeyValueConverterFactory(), logger, createHostSecretsIfMissing)
+        public SecretManager(ISecretsRepository repository, ILogger logger, IMetricsLogger metricsLogger, bool createHostSecretsIfMissing = false)
+            : this(repository, new DefaultKeyValueConverterFactory(), logger, metricsLogger, createHostSecretsIfMissing)
         {
         }
 
-        public SecretManager(ISecretsRepository repository, IKeyValueConverterFactory keyValueConverterFactory, ILogger logger, bool createHostSecretsIfMissing = false)
+        public SecretManager(ISecretsRepository repository, IKeyValueConverterFactory keyValueConverterFactory, ILogger logger, IMetricsLogger metricsLogger, bool createHostSecretsIfMissing = false)
         {
             _repository = repository;
             _keyValueConverterFactory = keyValueConverterFactory;
             _repository.SecretsChanged += OnSecretsChanged;
             _logger = logger;
+            _metricsLogger = metricsLogger;
+            _repositoryClassName = _repository.GetType().Name;
```

Comment:
throw ArgumentNullException here. Should probably add that for the other params as well to be consistent.